engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/ModificationExecutionAsyncTest.java

```diff
@@ -884,6 +1145,52 @@ public void testBatchCreationWithOverlappingProcessInstanceIdsAndQuery() {
     assertBatchCreated(batch, processInstanceCount);
   }
 
+  @Test
+  public void testBatchCreationWithOverlappingProcessInstanceIdsAndHistoricQuery() {
+    int processInstanceCount = 15;
+    DeploymentWithDefinitions deployment = testRule.deploy(instance);
+    ProcessDefinition processDefinition = deployment.getDeployedProcessDefinitions().get(0);
+    List<String> processInstanceIds = helper.startInstances("process1", 15);
+
+    HistoricProcessInstanceQuery historicProcessInstanceQuery = historyService.createHistoricProcessInstanceQuery().processDefinitionId(processDefinition.getId());
+    assertEquals(processInstanceCount, historicProcessInstanceQuery.count());
+
+    // when
+    Batch batch = runtimeService
+        .createModification(processDefinition.getId())
+        .startTransition("seq")
+        .processInstanceIds(processInstanceIds)
+        .historicProcessInstanceQuery(historicProcessInstanceQuery)
+        .executeAsync();
+
+    // then a batch is created
+    assertBatchCreated(batch, processInstanceCount);
+  }
+
+  @Test
+  public void testBatchCreationWithOverlappingHistoricQueryAndQuery() {
+    int processInstanceCount = 15;
+    DeploymentWithDefinitions deployment = testRule.deploy(instance);
+    ProcessDefinition processDefinition = deployment.getDeployedProcessDefinitions().get(0);
+    List<String> processInstanceIds = helper.startInstances("process1", 15);
```

Comment:
🙃 Not used.
```suggestion
```

Reply:
Removed `List<String> processInstanceIds =`.
`helper.startInstances("process1", 15);`   is required for the test